### DISCUSSION TITLE
Highlight atoms with gradient color circles

### DISCUSSION THREAD

**Discussion by peizhenbai** [rdkit/rdkit#4786] [Q&A]
Hi everyone,

I'd like to ask whether anyone knows how to draw the highlighted atoms with the gradient color just like the picture below. This is from one paper I read recently, which is a really beautiful visualization and I also want to achieve a similar effort in my project. I can get the result with DrawMolecule in RDKit but have no idea about the color change. Many thanks!

![image](https://user-images.githubusercontent.com/67964033/145477698-5a5f43cd-3cb8-450a-b336-4ab1f0d4abab.png)

**Comment by Sowmya-R-Krishnan** [comment]
Hi @pz-white,

I was trying a similar type of coloring scheme for one of my models. I used the following code to get the gradients for atoms based on a property value.

`pos_val = dict(sorted(pos_val.items(), key=lambda item: item[1], reverse=True))`

pos_val is a dictionary mapping a subset of atoms to their corresponding property values. I have sorted the dictionary by values in descending order. The mapping of a gradient of same color to the subset of atoms was done as follows:

```
def make_rgb_transparent(rgb, bg_rgb, alpha):
	return [alpha * c1 + (1 - alpha) * c2 for (c1, c2) in zip(rgb, bg_rgb)]

alpha = 1
rgb_pos = ColorConverter().to_rgb("red")
for atom, val in pos_val.items():
	rgb_new = make_rgb_transparent(rgb_pos, (1, 1, 1), alpha)
	alpha = alpha - 0.01
	highlight_dict[atom] = tuple(rgb_new)
	rgb_pos = rgb_new
```

The alpha value is used as in matplotlib to modify the transparency of the red color giving the sense of a gradient. Since the dictionary was sorted in descending order and I started with an alpha value of 1, the transparency decreases linearly with the property of interest with a step size of 0.01 (this was an arbitrary choice - the alpha value can be varied from 0 to 1 by any other method as well). The resultant color rgb_new was used to fill the highlight_dict dictionary, which maps the selected subset of atoms to the gradients of red color. I used this with rdMolDraw2D.PrepareAndDrawMolecule() to get an effect similar to your example image.
The make_rgb_transparent function was taken from this thread online: https://stackoverflow.com/questions/33371939/calculate-rgb-equivalent-of-base-colors-with-alpha-of-0-5-over-white-background

Hope this helps! If you have any other suggestions to do this more elegantly, please do let me know.
